sdk/identity/azure-identity/azure/identity/_credentials/chained.py

```diff
@@ -122,3 +125,64 @@ def get_token(
         )
         _LOGGER.warning(message)
         raise ClientAuthenticationError(message=message)
+
+    def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] = None) -> AccessTokenInfo:
+        """Request a token from each chained credential, in order, returning the first token received.
+
+        If no credential provides a token, raises :class:`azure.core.exceptions.ClientAuthenticationError`
+        with an error message from each credential.
+
+        This is an alternative to `get_token` to enable certain scenarios that require additional properties
+        on the token. This method is called automatically by Azure SDK clients.
+
+        :param str scopes: desired scopes for the access token. This method requires at least one scope.
+            For more information about scopes, see https://learn.microsoft.com/entra/identity-platform/scopes-oidc.
+        :keyword options: A dictionary of options for the token request. Unknown options will be ignored. Optional.
+        :paramtype options: ~azure.core.credentials.TokenRequestOptions
+
+        :rtype: AccessTokenInfo
+        :return: An AccessTokenInfo instance containing information about the token.
+
+        :raises ~azure.core.exceptions.ClientAuthenticationError: no credential in the chain provided a token.
+        """
+        within_credential_chain.set(True)
+        history = []
+        for credential in self.credentials:
+            try:
+                # A custom credential in the chain may not implement get_token_info
+                if hasattr(credential, "get_token_info"):
+                    token_info = cast(SupportsTokenInfo, credential).get_token_info(*scopes, options=options)
```

Comment:
It get_token_info is not implemented, we should skip it.

Reply:
If `get_token_info` is not implemented, we should still try to fall back to not break users who have chains with custom credentials that do not implement `get_token_info`.

A user might have the latest version of `azure-identity` installed, which will mean that `ChainedTokenCredential` will have `get_token_info` available. From the auth policies, `get_token_info` will be called on the passed-in `ChainedTokenCredential` instance, so as we iterate the chain on the identity side, we should fall back to avoid a user's credential suddenly being skipped when they update azure-identity.

```python

class MyCustomCredential:

    def get_token(self, *scopes, **kwargs):
        # custom logic here
        return AccessToken("token", expires_on=12345)


custom_chain = ChainedTokenCredential(
    ManagedIdentityCredential(),
    MyCustomCredential(),
    InteractiveBrowserCredential()
)

# MyCustomCredential should still be able to provide a token if ManagedIdentityCredential fails.
custom_chain.get_token_info('scope')

```

Reply:
But then get_pop_token will end up returning a non-pop token.

Reply:
This could be true even if a user calls `get_token_info` and the credential doesn't support PoP, right? From what I recall, the current design is that we will fall back to bearer token if the credential or resource doesn't support PoP.

Reply:
> we will fall back to bearer token if the credential or resource doesn't support PoP.

In current design, this should be handled by the policy.

We fall back to non-pop until we attempt all credentials and fail to get the pop.

In other words, the strategy is "we try our best to get a pop token"

Reply:
It is like (in the policy)
token = get_pop_token()
if token:
     auth header = "pop " + token
else:
    token = get_non_pop()
    auth header = "bearer " + token

Reply:
Let's do in this way:
if pop ("pop" in options) no fall back
else fall back

Reply:
Updated the logic here and added a couple test cases.